source/iostreams.tex

```diff
@@ -15315,6 +15318,100 @@
 Equivalent to: \tcode{return path(lhs) /= rhs;}
 \end{itemdescr}
 
+\rSec3[fs.path.fmtr]{Formatter support}
+
+\rSec4[fs.path.fmtr.general]{Formatter support overview}
+
+\indexlibraryglobal{formatter}%
+\begin{codeblock}
+namespace std {
+  template<class charT> struct formatter<filesystem::path, charT> {
+    constexpr void set_debug_format();
+
+    constexpr typename basic_format_parse_context<charT>::iterator
+      parse(basic_format_parse_context<charT>& ctx);
+
+    template<class FormatContext>
+      typename FormatContext::iterator
+        format(const filesystem::path& path, FormatContext& ctx) const;
+  };
+}
+\end{codeblock}
+
+\begin{ncbnf}
+\nontermdef{path-format-spec}\br
+    \opt{fill-and-align} \opt{width} \opt{\terminal{?}} \opt{\terminal{g}}
+\end{ncbnf}
+where the productions \fmtgrammarterm{fill-and-align} and \fmtgrammarterm{width}
+are described in \ref{format.string}.
+If the \tcode{?} option is used then
+the path is formatted as an escaped string\iref{format.string.escaped}.
+
+\rSec4[fs.path.fmtr.funcs]{Formatter support functions}
+
+\indexlibrarymember{formatter}{set_debug_format}%
+\begin{itemdecl}
+constexpr void set_debug_format();
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\effects
+Modifies the state of the \tcode{formatter} to be as if
+the \fmtgrammarterm{path-format-spec} parsed by the last call to \tcode{parse}
+contained the \tcode{?} option.
+\end{itemdescr}
+
+\indexlibrarymember{formatter}{basic_format_parse_context}%
+\begin{itemdecl}
+constexpr typename basic_format_parse_context<charT>::iterator
+  parse(basic_format_parse_context<charT>& ctx);
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\effects
+Parses the format specifier as a \fmtgrammarterm{path-format-spec} and
+stores the parsed specifiers in \tcode{*this}.
+
+\pnum
+\returns
+An iterator past the end of the \fmtgrammarterm{path-format-spec}.
+\end{itemdescr}
+
+\indexlibrarymember{formatter}{format}%
+\begin{itemdecl}
+template<class FormatContext>
+  typename FormatContext::iterator
+    format(const filesystem::path& p, FormatContext& ctx) const;
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\effects
+Let \tcode{s} be \tcode{p.generic_string<filesystem::path::value_type>()}
+if the \tcode{g} option is used,
+otherwise \tcode{p.native()}.
+Writes \tcode{s} into \tcode{ctx.out()},
+adjusted according to the \fmtgrammarterm{path-format-spec}.
+If \tcode{charT} is \keyword{char},
+\tcode{path::value_type} is \keyword{wchar_t}, and
+the literal encoding is UTF-8,
+then the escaped path is transcoded from the native encoding for
```

Comment:
I don't think "the escaped path" is correct here, but that's what the paper says so I'll ask LWG.

Reply:
This needs an LWG issue, the paper is wrong.

Reply:
Thanks! I'll leave this with you if that's OK?

Reply:
This is now [LWG 4070](https://cplusplus.github.io/LWG/issue4070).